core/include/drivers/clk_dt.h

```diff
@@ -0,0 +1,104 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2021, Bootlin
+ */
+
+#ifndef __DRIVERS_CLK_DT_H
+#define __DRIVERS_CLK_DT_H
+
+#include <kernel/dt.h>
+#include <stdint.h>
+#include <sys/queue.h>
+
+#define MAX_CLK_PHANDLE_ARGS 3
+
+/**
+ * struct clk_dt_phandle_args - Devicetree clock args
+ * @nodeoffset: Clock consumer node offset
+ * @args_count: Count of cells for the clock
+ * @args: Clocks consumer specifiers
+ */
+struct clk_dt_phandle_args {
+	int nodeoffset;
+	int args_count;
+	uint32_t *args;
+};
+
+/**
+ * struct clk_driver - clk driver setup struct
+ * setup:
+ */
+struct clk_driver {
+	TEE_Result (*setup)(const void *fdt, int nodeoffset);
```

Comment:
I think this callback could move to `struct dt_driver`, as a probe-from-dt-node callback reference.
Maybe in a later change...
Note that 'probe` maybe be a better name.

Reply:
Agreed, this would make sense to be in the dt_driver struct. The next serie I have regarding SCMI clocks uses the same mechanism with a generic DT probe. Would you mind waiting for the next serie to introduce that ?

Reply:
Ok. Let's see that later.
I played with your clk dt stuff and prepared also some dt_driver probing thing. Hence the many catches i reported.  I'll also create a RFC for my proposal, once your changes are accepted. We discuss by then. @ldebieve you may be interested by these proposals.

Reply:
Ok great ! I'll see if it fits what I need but I guess so.